pkg/apis/workloads/v1alpha1/console_types.go

```diff
@@ -120,6 +120,37 @@ type ConsoleTemplateSpec struct {
 	// +kubebuilder:validation:Minimum=0
 	// +kubebuilder:validation:Maximum=604800
 	DefaultTTLSecondsAfterFinished *int32 `json:"defaultTtlSecondsAfterFinished,omitempty"`
+
+	// List of authorisation rules to match against in order from top to bottom.
+	// +optional
+	AuthorisationRules []ConsoleAuthorisationRule `json:"authorisationRules,omitempty"`
+	// Default authorisation rule to use if no authorisationRules are defined or no authorisationRules match.
+	// +optional
+	DefaultAuthorisationRule *ConsoleDefaultAuthorisationRule `json:"defaultAuthorisationRule,omitempty"`
+}
+
+// ConsoleAuthorisationRule declares rules specifying what commands need to be authorised and by whom.
+type ConsoleAuthorisationRule struct {
+	Name string `json:"name"`
+
+	// The match expression to compare to the command and arguments of the console.
+	// Uses the glob format https://github.com/gobwas/glob.
+	MatchCommand string `json:"matchCommand"`
+
+	// The number of authorisations required from members of the subjects before the console can run.
+	AuthorisationsRequired int `json:"authorisationsRequired"`
```

Comment:
What strikes me here is that we've duplicated these two fields between `ConsoleAuthorisationRule` and `ConsoleDefaultAuthorisationRule`.

Is it possible for these to be extracted out into a struct, which we embed within `ConsoleAuthorisationRule`? (And if we did that, we might want to change the naming a bit).

I can't remember if it's doable in this world with all of the codegen, or not.

If we can do it then I think it might make our lives simpler when coming to write the reconciliation logic, as we'll probably need to be appending the default rule onto the end of the list of rules in order to process them, so it may make sense to use the same types.

Reply:
Going to extract this into a struct called `ConsoleAuthorisers` and see how it goes.

Reply:
FYI had to add a json tag (`` `json:",inline"` ``) to the embedded struct for the codegen to work as expected.